zebra/zebra_rib.c

```diff
@@ -1565,7 +1574,7 @@ static bool rib_compare_routes(const struct route_entry *re1,
 	 * v6 link-locals, and we also support multiple addresses in the same
 	 * subnet on a single interface.
 	 */
-	if (re1->type != ZEBRA_ROUTE_CONNECT)
+	if (re1->type != ZEBRA_ROUTE_CONNECT && re1->type != ZEBRA_ROUTE_LOCAL)
```

Comment:
Not sure `L` is equivalent to `C` in handling for this specific thing. Can we have duplicate `L` routes on the same interface?

Also: this specifically needs to be documented since it's a design decision. e.g. in the "same ip address with different prefix length". Do we have one or two `L` routes?

Reply:
this behavior is correct in that we already only select the first one

Reply:
⇒ two L routes ⇒ code is correct as is ⇒ closing